shell/common/animator.cc

```diff
@@ -107,6 +123,47 @@ void Animator::BeginFrame(fml::TimePoint frame_start_time,
     TRACE_FLOW_END("flutter", "PointerEvent", trace_flow_id);
     trace_flow_ids_.pop_front();
   }
+  while (!pointer_datas_.empty()) {
+    const PointerData& pointer_data = pointer_datas_.front();
+    switch (pointer_data.change) {
+      case flutter::PointerData::Change::kDown: {
+        PointerDelta delta;
+        delta.physical_x = pointer_data.physical_x;
+        delta.physical_y = pointer_data.physical_y;
+        delta.last_physical_x = pointer_data.physical_x;
+        delta.last_physical_y = pointer_data.physical_y;
+        down_pointers_[pointer_data.device] = delta;
+      } break;
+      case flutter::PointerData::Change::kCancel:
+      case flutter::PointerData::Change::kUp:
+        down_pointers_.erase(pointer_data.device);
+        break;
+      case flutter::PointerData::Change::kMove: {
+        auto it = down_pointers_.find(pointer_data.device);
+        if (it != down_pointers_.end()) {
+          it->second.physical_x = pointer_data.physical_x;
+          it->second.physical_y = pointer_data.physical_y;
+        }
```

Comment:
nit: add `FML_DCHECK(it != down_pointers_.end())` as `kMove` should always follow `kDown`?